src/Microsoft.AspNet.OData/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -27,6 +28,12 @@ public HttpRequestMessage Request
                 _request = value;
                 InternalRequest = _request != null ? new WebApiRequestMessage(_request) : null;
                 InternalUrlHelper = _request != null ? new WebApiUrlHelper(_request.GetUrlHelper()) : null;
+
+                var webApiRequestMessage = (WebApiRequestMessage)InternalRequest;
```

Comment:
>var webApiRequestMessage = (WebApiRequestMessage)InternalRequest; [](start = 16, length = 65)

We should be able to avoid a cast here.  Rather than set InternalRequest as a WebApiRequestMessage in line 39 and then cast back to WebApiRequestMessage in line 32, we should set webApiRequestMessage as new WebApiRequestMessage(_request) and then set InternalRequest to be that.

Reply:
#Resolved